### PR TITLE
ui/state: Deduplicate query results that were being stored in state slices

### PR DESCRIPTION
**Pull request by ssangervasi**
Related to #1075 

### Code Changes

- Unused state:
    - Systems
    - Datasets
    - Organization hydrate
- Queries with duplicate state:
    - Taxonomy DataCategories query
    - DataSubjects query
    - DataQualifiers query
    - DataUses query

### Steps to Confirm

* [ ] Config wizard should work the same

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* Documentation Updated:
  - [ ] documentation complete, or draft/outline provided (tag docs-team to complete/review on this branch)
  - [ ] documentation issue created (tag docs-team to complete issue separately)
* [ ] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [ ] Update `CHANGELOG.md`

### Description Of Changes

This does away with some state duplication that has sprung up due to some confusion about RTK Query vs slices.

**TLDR**: Use `useFooQuery` and/or `createSelector(fooApi.endpoints.fooQuery.select(), ...)`. Do not re-store query results.

With this PR I think the pattern should be easier to follow going forward, and this will live as a bit of documentation. I'll comment explaining on individual lines changed. First, an overview:

- `createApi` already stores query results in the Redux state.
    - This is referred to as the "cache".
    - [RTK provides a helper](https://redux-toolkit.js.org/rtk-query/usage/usage-without-react-hooks#accessing-cached-data--request-status) for creating a selector: `api.endpoints.fooQuery.select()`
    - This will only be populated if the query has run. The hooks are still useful!
    - The cache also takes into account the parameters of the request. This PR only has examples where the request takes no params.
- `useQuery` [subscribes](https://redux-toolkit.js.org/rtk-query/api/created-api/hooks#usequery) to the API _and_ provides access to the cache.
    - For any query, you **need at least one hook call** to actually populate the cache.
    - The response `data` is the same as the value in the cache.
- `selectFromResult` is a [shortcut](https://redux-toolkit.js.org/rtk-query/usage/queries#selecting-data-from-a-query-result) for subscribing _and_ deriving a value from the cache.
    - This is handy for one-off query hook uses, but if the derived value is shared then the logic should be in its own selector.
    - [Example use](https://github.com/ethyca/fides/blob/dc8b6f3ee9889aac4039ad180e9e5601706c4bdf/clients/ctl/admin-ui/src/features/common/plus.slice.ts#L89)
- `createSlice` is mainly for user-editable state.
    - If calling `setFoo` only ever happens once, it probably shouldn't happen at all.
    - There are some empty slices now that the query results have been removed. If it turns out we won't need any user input in these featuers, we can remove those entirely.
    - You might still find a use case for storing a (part of a) query result in the state: if the user can edit the object and that state needs to live outside of a single component/form.

### CODE REVIEW

File: clients/ctl/admin-ui/src/features/config-wizard/PrivacyDeclarationForm.tsx [hunk 1]
```diff
@@ -76,25 +72,17 @@ const PrivacyDeclarationForm = ({
   const [updateSystem] = useUpdateSystemMutation();
   const [isLoading, setIsLoading] = useState(false);
 
-  const { data: dataCategories } = useGetAllDataCategoriesQuery();
-  const { data: dataSubjects } = useGetAllDataSubjectsQuery();
-  const { data: dataQualifiers } = useGetAllDataQualifiersQuery();
-  const { data: dataUses } = useGetAllDataUsesQuery();
+  // Query subscriptions:
```
Comment (ssangervasi) [inline clients/ctl/admin-ui/src/features/config-wizard/PrivacyDeclarationForm.tsx:75]: Note that we still need the hook calls to subscribe to the endpoint, we just don't need to manually store the results in the state. For resources that are used all over the app, we could move the subscription closer to the root of the app and/or [increase cache time](https://redux-toolkit.js.org/rtk-query/usage/cache-behavior#cache-lifetime--subscription-example) .

File: clients/ctl/admin-ui/src/features/data-qualifier/data-qualifier.slice.ts [hunk 2]
```diff
@@ -64,19 +56,20 @@ export const {
...
-export const { setDataQualifiers } = dataQualifierSlice.actions;
-export const selectDataQualifiers = (state: AppState) =>
-  state.dataQualifier.dataQualifier;
-
 export const { reducer } = dataQualifierSlice;
+
+const emptyDataQualifiers: DataQualifier[] = [];
+export const selectDataQualifiers: (state: AppState) => DataQualifier[] =
+  createSelector(
+    dataQualifierApi.endpoints.getAllDataQualifiers.select(),
```
Comment (ssangervasi) [inline clients/ctl/admin-ui/src/features/data-qualifier/data-qualifier.slice.ts:73]: This returns a function. I explicitly wrote the return type `(state: AppState) => DataQualifier[]` for these selectors, but the type inference of `createSelector` and `api.endpoints.foo.select` actually does the trick on its own.

File: clients/ctl/admin-ui/src/features/data-qualifier/data-qualifier.slice.ts [hunk 3]
```diff
@@ -64,19 +56,20 @@ export const {
...
+  reducers: {},
 });
 
-export const { setDataQualifiers } = dataQualifierSlice.actions;
-export const selectDataQualifiers = (state: AppState) =>
-  state.dataQualifier.dataQualifier;
-
 export const { reducer } = dataQualifierSlice;
+
+const emptyDataQualifiers: DataQualifier[] = [];
```
Comment (ssangervasi) [inline clients/ctl/admin-ui/src/features/data-qualifier/data-qualifier.slice.ts:70]: As a reminder: creating a constant for an empty selection results reduces some re-rendering by preserving the selector's memoization. Not important for performance at the moment, but it's a helpful habit.

Comment (allisonking) [reply clients/ctl/admin-ui/src/features/data-qualifier/data-qualifier.slice.ts:70]: thanks for the explanation! did not know

**Review by allisonking — APPROVED** [review]
amazing! thank you for diving into this and for the super clear explanations 🙏

**Review by LKCSmith — APPROVED** [review]
🔥
